Source/Particles/Collision/BinaryCollision/ParticleCreationFunc.H

```diff
@@ -115,7 +115,7 @@ public:
     {
         using namespace amrex::literals;
 
-        if (n_total_pairs == 0) return {m_num_product_species, 0};
```

Comment:
Our intent was to return an `amrex::Vector<int>` of size `m_num_product_species` filled with `0`, but instead this actually returns an `amrex::Vector<int>` of size `2`, filled with `[ m_num_product_species, 0 ]`.

Reply:
Interesting, this complains about it:
```
Source/Particles/Collision/BinaryCollision/ParticleCreationFunc.H:118:40: warning: avoid repeating the return type from the declaration; use a braced initializer list instead [modernize-return-braced-init-list]
        if (n_total_pairs == 0) return amrex::Vector<int>(m_num_product_species, 0);
                                       ^
```

@lucafedeli88 I think #4133 changed this by accident.

Reply:
Oh, ok. But then I am not sure how get the correct behavior, while still making `clang-tidy` happy. 
Any suggestion @ax3l @lucafedeli88 ? As a reminder, using the following braced initializer list does not do what we want: 
```
return {m_num_product_species, 0}
```

Reply:
ah! I see... that's true... I am very sorry. 
I blindly followed this suggestion, which is misleading in this case
```
2023-10-03T19:15:46.7620300Z /home/runner/work/WarpX/WarpX/Source/Particles/Collision/BinaryCollision/ParticleCreationFunc.H:118:40: warning: avoid repeating the return type from the declaration; use a braced initializer list instead [modernize-return-braced-init-list]
2023-10-03T19:15:46.7621127Z         if (n_total_pairs == 0) return amrex::Vector<int>(m_num_product_species, 0);
```

Reply:
@lucafedeli88 I think this test has a bug and does not exclude transformations of non-init-list constructors.
[Looking at the tests here](https://reviews.llvm.org/D28768), we might want to:
- exclude the test for now
- report a bug with LLVM

Reply:
I agree

Reply:
@lucafedeli88 I'll file the LLVM bug.

Reply:
@lucafedeli88 filed in https://github.com/llvm/llvm-project/issues/68159